### ISSUE TITLE
Support non-string formula?

### ISSUE THREAD

**Issue by multimeric** [n1m3/linregress#7]
I'm interesting in running regression on the entire set of variables. However it seems that the `FormulaRegressionBuilder` requires a formula to work. Is there any thought to adding an "all variables" mode, or else provide an array of variable names so that it's a bit easier to automate? I would like to avoid building a string formula programatically if I can.

**Comment by n1m3** [comment]
Hmm yes, the library seems a bit too closely tied to the formula idea.

I've added a [`data_columns`](https://github.com/n1m3/linregress/blob/ababcd69ebcb9ceb4594e2c84e2054f69b7ac5a3/src/lib.rs#L178) method to `FormulaRegressionBuilder` on the master branch.
Usage would look like this:
```rust
let data = vec![("Y", y), ("X1", x1), ("X2", x2), ("X3", x3)];
let data = RegressionDataBuilder::new().build_from(data).unwrap();
let model = FormulaRegressionBuilder::new()
    .data(&data)
    .data_columns("Y", ["X1", "X2", "X3"])
    .fit()
    .unwrap();
```
Does that work for your use case?

**Comment by multimeric** [comment]
That's an improvement. But actually I want to support regression with a big unlabelled matrix (`ndarray`), which doesn't even have vector names. So I would still have to generate names for each vector, but at least that would be easier than building a big string.

**Comment by n1m3** [comment]
Ah, I see.
How about [this](https://github.com/n1m3/linregress/blob/96a5cd02ed1d27f58325d081729e492a4cf0f0a4/src/lib.rs#L904) then?
See doc comment for intended usage.
I'm not really willing to expose `nalgebra` types in the API, so that's about as low level, as I can reasonably offer.

**Comment by multimeric** [comment]
Yep I think that would be fine. I'm sure there's an `ndarray` function to unwrap into a row-major array. Thanks!

**Comment by n1m3** [comment]
The new APIs are now available in the [new version `0.4.4`](https://crates.io/crates/linregress/0.4.4)
